### PR TITLE
[fix]: [CDS-73527]: Made validate-template-input public with VIEW RBAC permissions.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes: 
Context: A public api is required by customers to reconcile pipeline

Change: Made the already existing "validate-template-inputs" api public.

Impact: Reconcile API.

Testing: The required api was observed on the openApiSpec and earlier it was not observed there.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger uts

**Review by utkarshchoubey-harness — APPROVED** [review]

**Review by ankittiwari-harness — APPROVED** [review]

**Comment by senjucanon2** [comment]
trigger review

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 1 file(s) to review. 1 change requests. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 115 at r1](https://reviewable.io/reviews/harness/harness-core/51048#-NaL7B434NP_YpoBVQqR:-NaL7B434NP_YpoBVQqS:byb10at) ([raw file](https://github.com/harness/harness-core/blob/2ad80129ac2a47cb7abf5ce8adba5b57c08da797/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L115)):*
> ```Java
>   @GET
>   @Path("validate-template-inputs")
>   @ApiOperation(value = "Validates whether yaml of pipeline is valid or not", nickname = "validateTemplateInputs")
> ```

If we are making API public, we should get approval from docs team as well on the API spec details.

Personally, I'm not able to interpret the descriptions correctly from the ones mentioned right now. We should improve on it.

**Comment by senjucanon2** [comment]
trigger uts

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 68 at r3](https://reviewable.io/reviews/harness/harness-core/51048#-NaQKiLQ7rRfWPd8nLH0:-NaQKiLQ7rRfWPd8nLH1:bli9npd) ([raw file](https://github.com/harness/harness-core/blob/008fc8ea071a12f39cca98bb31e5b30f416007a2/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L68)):*
> ```Java
>       , @ApiResponse(code = 500, response = ErrorDTO.class, message = "Internal server error")
>     })
> @Tag(name = "Pipeline", description = "This contains APIs related to validation of templates")
> ```

The description looks incorrect for the tag name provided.

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 115 at r1](https://reviewable.io/reviews/harness/harness-core/51048#-NaL7B434NP_YpoBVQqR:-NaQY-hrATeWaa3ZnZaf:bo212nu) ([raw file](https://github.com/harness/harness-core/blob/2ad80129ac2a47cb7abf5ce8adba5b57c08da797/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L115)):*
<details><summary><i>Previously, senjucanon2 (Vivek Dixit) wrote…</i></summary><blockquote>

This ticket is raised by PM, whom else approval is needed?
</blockquote></details>

Have improvised description for fields in Api.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 68 at r3](https://reviewable.io/reviews/harness/harness-core/51048#-NaQKiLQ7rRfWPd8nLH0:-NaQXs4Y1o-RjKwIeIwJ:bz6ufkj) ([raw file](https://github.com/harness/harness-core/blob/008fc8ea071a12f39cca98bb31e5b30f416007a2/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L68)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

The description looks incorrect for the tag name provided.
</blockquote></details>

Have Improvised tag name and description.

File: pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java [hunk 1]
```diff
@@ -110,11 +114,19 @@ public class PipelineRefreshResource {
...
   @Path("validate-template-inputs")
-  @ApiOperation(value = "This validates whether yaml of pipeline is valid or not", nickname = "validateTemplateInputs")
-  @Hidden
-  public ResponseDTO<ValidateTemplateInputsResponseDTO> validateTemplateInputs(
-      @Parameter(description = NGCommonEntityConstants.ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
-          NGCommonEntityConstants.ACCOUNT_KEY) @AccountIdentifier String accountId,
+  @ApiOperation(value = "Validates whether yaml of pipeline is valid or not", nickname = "validateTemplateInputs")
+  @NGAccessControlCheck(resourceType = "PIPELINE", permission = PipelineRbacPermissions.PIPELINE_VIEW)
+  @Operation(operationId = "validateTemplateInputs",
+      description = "Validates pipeline yaml for template inputs, and if invalid returns error summary",
```
Comment (deepakbolangady) [inline pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java:120]:
```suggestion
      description = "Validates the template inputs in a pipeline's YAML specification. If the template inputs are invalid, the operation returns an error summary.",
```

File: pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java [hunk 2]
```diff
@@ -110,11 +114,19 @@ public class PipelineRefreshResource {
...
+      summary = "Validates template inputs in pipeline yaml",
```
Comment (deepakbolangady) [inline pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java:121]:
```suggestion
      summary = "Validates template inputs in a pipeline's YAML specification.",
```

File: pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java [hunk 3]
```diff
@@ -110,11 +114,19 @@ public class PipelineRefreshResource {
...
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "default",
+            description = "Returns pipeline yaml validation result for template inputs present in pipeline yaml")
```
Comment (deepakbolangady) [inline pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java:125]:
```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```

Comment (deepakbolangady) [inline pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java:125]:
```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```

Comment (deepakbolangady) [inline pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java:125]:
```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```

**Review by deepakbolangady — APPROVED** [review]
Approving with the assumption that you will look into these suggestions. Thanks.

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 1 file(s) to review. 1 change requests. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 117 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277398651:-NaeCAFM5SQOzZna8SbL:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L117)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
  @ApiOperation(value = "Validates the YAML specification of a pipeline.", nickname = "validateTemplateInputs")
```
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 120 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277414607:-NaeCAaN0qFk6jxLXOfy:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L120)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
      description = "Validates the template inputs in a pipeline's YAML specification. If the template inputs are invalid, the operation returns an error summary.",
```
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 121 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277415345:-NaeCBtRDquFMb1639c5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L121)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
      summary = "Validates template inputs in a pipeline's YAML specification.",
```
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 125 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277426659:-NaeCCFN6yL1kUqMXtgS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L125)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 125 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277427217:-NaeCCjU6E6A5Scx7exU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L125)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java` line 125 at r4](https://reviewable.io/reviews/harness/harness-core/51048#gh-1277427239:-NaeCDAz6m2E679qMfbb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ac238bf84495b60c7e455d07eab4ab67301851c3/pipeline-service/service/src/main/java/io/harness/pms/template/PipelineRefreshResource.java#L125)):*
<details><summary><i>Previously, deepakbolangady (Deepak Bolangady) wrote…</i></summary><blockquote>

```suggestion
            description = "Returns a validation result for template inputs present in a pipeline's YAML specification.")
```
</blockquote></details>

Done.

**Comment by senjucanon2** [comment]
trigger uts

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 1 file(s) to review. 6. unresolved discussions. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger uts

**Review by archit-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51048)* status: 2 file(s) to review. 6. unresolved discussions. 0 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]

**Comment by senjucanon2** [comment]
trigger review

**Review by shalini-agr — APPROVED** [review]

**Comment by senjucanon2** [comment]
trigger review

**Pull request merged**